app/Http/Controllers/Inertia/TransactionsController.php

```diff
@@ -4,22 +4,102 @@
 
 namespace App\Http\Controllers\Inertia;
 
+use App\DTO\Inertia\Transaction as TransactionDTO;
+use App\Http\Controllers\Inertia\Concerns\WithFilters;
+use App\Http\Controllers\Inertia\Concerns\WithPagination;
+use App\Models\Scopes\OrderByTimestampScope;
+use App\Models\Scopes\OrderByTransactionIndexScope;
+use App\Models\Transaction;
 use App\Services\BigNumber;
 use App\Services\Cache\StatisticsCache;
+use ARKEcosystem\Foundation\UserInterface\UI;
+use Illuminate\Pagination\LengthAwarePaginator;
 use Inertia\Inertia;
 use Inertia\Response;
 
 final class TransactionsController
 {
+    use WithPagination;
+    use WithFilters;
+
+    public const FILTERS = [
+        'transfers'           => true,
+        'multipayments'       => true,
+        'votes'               => true,
+        'validator'           => true,
+        'username'            => true,
+        'contract_deployment' => true,
+        'others'              => true,
+    ];
+
     public function __invoke(): Response
     {
-        $data = (new StatisticsCache())->getTransactionData();
-
         return Inertia::render('Transactions/Transactions', [
-            'transactionCount' => $data['transaction_count'],
-            'volume'           => BigNumber::new($data['volume'])->toFloat(),
-            'totalFees'        => BigNumber::new($data['total_fees'])->toFloat(),
-            'averageFee'       => BigNumber::new($data['average_fee'])->toFloat(),
+            'filters'          => fn () => $this->filters(),
```

Comment:
Filters aren't working correctly right now. You may not need to do it like the below. The filters from the controller are used as an initial value for the React filter component, so it may confuse the React side of things.

```suggestion
            'filters'          => self::FILTERS,
```

If you do change this, you can also probably remove filters from requests - `only: ["transactions"]`